wiggum_app/controller.py

```diff
@@ -235,22 +234,35 @@ def main():
             filter_result = labeled_df_setup.get_trend_rows(feat1=filter_object['feat1'],feat2=filter_object['feat2'],
                                 group_feat=filter_object['group_feat'],subgroup=filter_object['subgroup'],
                                 trend_type =filter_object['trend_type'])
+            
+            # Generate distance heatmaps
+            distance_heatmap_dict = models.getDistanceHeatmapDict(filter_result)
+
+            df = labeled_df_setup.df.to_dict(orient='records')
+            df = json.dumps(df, indent=2)
 
-            result_dict_dict = {}
-            result_dict_dict = models.getResultDict(labeled_df_setup, filter_result, filter_object['subgroup'])
+            #return jsonify(result_dict_dict)
+            return jsonify(distance_heatmap_dict = distance_heatmap_dict, 
+                            result_df = filter_result.to_json(orient='records'),
+                            df = df)
 
-            return jsonify(result_dict_dict)
 
         # visualize.html 'Reset' button clicked
         if action == 'reset':
```

Comment:
Can you add a few lines of comments at each of these if cases that outlines what is done?